SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/EventTracingTests/EventTracingTest.cpp

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "EventTracingTest.h"
+
+#include "EngineClasses/SpatialGameInstance.h"
+#include "Interop/Connection/SpatialConnectionManager.h"
+#include "Interop/Connection/SpatialEventTracer.h"
+
+#include "SpatialFunctionalTestFlowController.h"
+
+#include <WorkerSDK/improbable/c_io.h>
+#include <WorkerSDK/improbable/c_trace.h>
+
+using namespace SpatialGDK;
+using namespace worker::c;
+
+AEventTracingTest::AEventTracingTest()
+{
+	Author = "Matthew Sandford";
+	Description = TEXT("Base class for event tracing tests");
+
+	SetNumRequiredClients(1);
+}
+
+void AEventTracingTest::BeginPlay()
+{
+	Super::BeginPlay();
+
+	TestStartTime = FDateTime::Now();
+
+	{
```

Comment:
These are probably not necessary for you, I only had them in some tests due to certain variable redefinitions.

Reply:
Yeah tru